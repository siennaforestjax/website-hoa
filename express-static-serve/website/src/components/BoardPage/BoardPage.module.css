:root {
  --boardMemberPaddingBottom: 30px;
}

.boardMember {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 25px;
  width: 100%;
  padding-bottom: var(--boardMemberPaddingBottom);
  justify-content: center;
}

.figure {
  min-width: 200px;
  min-height: 225px;
  max-width: 200px;
  max-height: 225px;
}

.figure > * {
  margin: auto;
}

.figureCaption {
  font-size: 14px;
  font-style: italic;
  text-align: center;
}

.details {
  margin-top: 10px;
  padding-left: 15px;
}

.details p:not(:last-of-type) {
  padding-bottom: 4px;
}

.fieldDescription {
  text-decoration: underline;
}

.fieldTitle {
  font-weight: bold;
}

.fieldValue {
  display: inline-block;
  padding-left: 5px;
}

.subcommitteeHeader {
  margin-top: 25px;
}

.subcommitteeSection {
  width: 100%;
  display: flex;
  flex-direction: row;
}

.subcommittee {
  margin-top: 25px;
  width: 80%;
  padding-bottom: var(--boardMemberPaddingBottom);
}

.subcommittee:last-of-type {
  border: none;
  padding-bottom: 0;
}

.committeeName {
  font-weight: bold;
}

.objectiveList {
  display: block;
  margin-top: 25px;
}

.subsection {
  margin-top: 25px;
}

.objective {
  margin-left: 30px;
  text-transform: capitalize;
}

.committeeMember {
  display: flex;
  text-transform: capitalize;
  column-gap: 25px;
  margin-left: 30px;
  width: 30%;
}

.row {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
}

.headerRow {
  font-weight: bold;
}

.columnTwo {
  flex-grow: 1;
}

.imageWrapper {
  width: 100%;
  height: calc(100% - 20px);
}

/*------Medium Screens--------*/
@media screen and (min-width: 600px) {
  :root {
    --detailsWidth: 35%;
  }

  .boardMember {
    flex-wrap: nowrap;
  }

  .details {
    width: calc(100% - var(--detailsWidth));
    margin-top: 0;
  }

  .subcommitteeSection {
    padding-left: 15px;
  }
}

@media screen and (min-width: 360px) and (max-width: 599px) {
  :root {
    --detailsWidth: 80vw;
  }

  .details {
    width: var(--detailsWidth);
    margin-top: 0;
    padding-left: 0;
  }

  .figure {
    min-width: 250px;
    min-height: 250px;
    max-width: 314px;
    max-height: 314px;
  }

  .boardMember {
    row-gap: 15px;
  }
}
