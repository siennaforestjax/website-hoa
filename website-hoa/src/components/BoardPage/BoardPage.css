.boardMember {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 25px;
  width: 100%;
  padding-bottom: 20px;
}

figure {
  width: 100%;
  height: 220px;
}

figure > * {
  margin: auto;
}

figcaption {
  font-size: 12px;
  font-style: italic;
  text-align: center;
}

img {
  border: 2px solid rgb(var(--primaryColor));
  border-radius: 21%;
  height: calc(100% - 20px);
  width: 200px;
}

.details {
  margin-top: 10px;
  padding-left: 15px;
  flex-basis: 1;
}

.details p:not(:last-of-type) {
  margin-bottom: 4px;
}

.fieldDescription {
  text-decoration: underline;
}

.fieldTitle {
  font-weight: bold;
}
