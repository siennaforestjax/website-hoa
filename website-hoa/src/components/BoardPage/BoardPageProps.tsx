import BoardMember from '../../interfaces/BoardMember';

export type BoardPageProps = {
  boardMembers: Array<BoardMember>;
};
