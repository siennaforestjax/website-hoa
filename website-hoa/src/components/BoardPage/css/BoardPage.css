.boardMember {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 25px;
  width: 100%;
  padding-bottom: 20px;
}

figure {
  width: 100%;
  height: 220px;
}

figure > * {
  margin: auto;
}

figcaption {
  font-size: 12px;
  font-style: italic;
  text-align: center;
}

img {
  height: calc(100% - 20px);
  min-width: 170px;
  min-height: 170px;
  max-width: 250px;
  max-height: 250px;
  border: 2px solid rgb(var(--primaryColor));
  border-radius: 21%;

  margin: auto;
}

.details {
  margin-top: 10px;
  padding-left: 15px;
}

.details p:not(:last-of-type) {
  margin-bottom: 4px;
}

.fieldDescription {
  text-decoration: underline;
}

.fieldTitle {
  font-weight: bold;
}
