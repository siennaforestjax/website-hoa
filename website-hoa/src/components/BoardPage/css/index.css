@import './BoardPage.css';
@import './BoardPage-med.css';
