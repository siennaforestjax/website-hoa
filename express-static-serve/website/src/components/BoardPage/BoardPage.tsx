import React from 'react';
import classes from './BoardPage.module.css';
import { BoardPageProps } from './BoardPageProps';

const BoardPage = ({ boardMembers }: BoardPageProps) => {
  return (
    <div className="pageWrapper">
      <h1>Meet The Board</h1>
      {boardMembers.map((boardMember) => {
        return (
          <section
            className={classes.boardMember + ' colorful-bottom-border'}
            key={boardMember.title}
          >
            <figure className={classes.figure}>
              <img
                className={classes.image}
                src={boardMember.imageName}
                alt={`${boardMember.name} (${boardMember.title})`}
              ></img>
              <figcaption className={classes.figureCaption}>
                {boardMember.imageDescription}
              </figcaption>
            </figure>
            <div className={classes.details}>
              <p>
                <span className="fieldTitle">{boardMember.title}</span> -{' '}
                {boardMember.name}
              </p>

              <p>{boardMember.titleDescription}</p>
            </div>
          </section>
        );
      })}
    </div>
  );
};

export default BoardPage;
