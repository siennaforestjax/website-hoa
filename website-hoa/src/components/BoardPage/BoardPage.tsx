import './BoardPage.css';
import React from 'react';
import { BoardPageProps } from './BoardPageProps';

const BoardPage = ({ boardMembers }: BoardPageProps) => {
  return (
    <div className="pageWrapper">
      <h1>Meet The Board</h1>
      {boardMembers.map((boardMember) => {
        return (
          <section
            className="boardMember colorful-bottom-border"
            key={boardMember.title}
          >
            <figure>
              <img
                src={boardMember.imageName}
                alt={`Photo of ${boardMember.name} (${boardMember.title})`}
              ></img>
              <figcaption aria-hidden>
                {boardMember.imageDescription}
              </figcaption>
            </figure>
            <div className="details">
              <p>
                <span className="fieldTitle">{boardMember.title}</span> -{' '}
                {boardMember.name}
              </p>

              <p>{boardMember.titleDescription}</p>
            </div>
          </section>
        );
      })}
    </div>
  );
};

export default BoardPage;
