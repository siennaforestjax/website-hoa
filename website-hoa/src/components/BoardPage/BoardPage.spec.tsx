import React from 'react';
import BoardPage from './BoardPage';
import renderer from 'react-test-renderer';
import testData from '../../boardMembers.json';

describe('The Board Page', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(<BoardPage boardMembers={testData.boardMembers} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
