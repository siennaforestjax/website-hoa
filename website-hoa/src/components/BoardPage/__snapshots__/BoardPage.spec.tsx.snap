// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Board Page The Snapshot renders as expected 1`] = `
<div
  className="pageWrapper"
>
  <h1>
    Meet The Board
  </h1>
  <section
    className="boardMember colorful-bottom-border"
  >
    <figure>
      <img
        alt="Photo of Kurt Ringo (President)"
        src="president.jpg"
      />
      <figcaption
        aria-hidden={true}
      >
        too cool for school
      </figcaption>
    </figure>
    <div
      className="details"
    >
      <p>
        <span
          className="fieldTitle"
        >
          President
        </span>
         -
         
        Kurt Ringo
      </p>
      <p>
        The HOA President serves as the leader of the board. The main responsibilities of the president include presiding over board meetings and facilitating the decision-making process; signing contacts, checks, and other legal documents; serving as a spokesman for the HOA; and taking charge of the day-to-day operations.
      </p>
    </div>
  </section>
  <section
    className="boardMember colorful-bottom-border"
  >
    <figure>
      <img
        alt="Photo of Maxine Holcomb (Treasurer)"
        src="treasurer.jpg"
      />
      <figcaption
        aria-hidden={true}
      >
        growing up on a farm
      </figcaption>
    </figure>
    <div
      className="details"
    >
      <p>
        <span
          className="fieldTitle"
        >
          Treasurer
        </span>
         -
         
        Maxine Holcomb
      </p>
      <p>
        The HOA Treasurer oversees the financial operations of the association.  They are in charge of developing the budget, monitoring income and expenses, collecting assessments, issuing payments to vendors, preparing financial reports, and facilitating an audit at the end of a fiscal year.
      </p>
    </div>
  </section>
  <section
    className="boardMember colorful-bottom-border"
  >
    <figure>
      <img
        alt="Photo of Caroline Shank (Secretary)"
        src="secretary.jpg"
      />
      <figcaption
        aria-hidden={true}
      >
        scary hair
      </figcaption>
    </figure>
    <div
      className="details"
    >
      <p>
        <span
          className="fieldTitle"
        >
          Secretary
        </span>
         -
         
        Caroline Shank
      </p>
      <p>
        The HOA secretary creates the board meeting agenda (with the president), records the minutes of board meetings, and ensures quorum. They also send notices for board meetings, ensure compliance with the governing documents, maintain bank information and other important documents, and store homeowner and maintenance records.
      </p>
    </div>
  </section>
</div>
`;
